files/class-api-client.php

```diff
@@ -219,6 +226,9 @@ public function is_file( $file_path, &$info = null ) {
 			$response_body = wp_remote_retrieve_body( $response );
 			$info = json_decode( $response_body, true );
 
+			// cache file info
+			$this->cache->cache_file_stats( $file_path, $info );
```

Comment:
Would we also want to cache failures (404s), or is that overkill?

Reply:
I think it's overkill. I can't think of a situation where that would be necessary.